pkg/traceql/engine.go

```diff
@@ -261,7 +261,7 @@ func (e *Engine) asTraceSearchMetadata(spanset *Spanset) *tempopb.TraceSearchMet
 		atts := span.AllAttributes()
 
 		if name, ok := atts[NewIntrinsic(IntrinsicName)]; ok {
-			tempopbSpan.Name = name.S
+			tempopbSpan.Name = name.EncodeToString(false)
```

Comment:
Isn't this changing behaviour? Before it'd call `.String()` which added quotes.

Reply:
Both `name.S` and `name.name.EncodeToString(false)` return the unquoted string. Therefore the behavior remains the same

Reply:
BTW: I find it quite confusing that `Static.String()` returns the quoted string. Without looking at the actual implementation I'd expect the following condition to be true:
```go
str := "foo"
if str == NewStaticString(str).String() {
    fmt.Println("works")
}
```
But it currently isn't. 

I believe it would improve the readability of the code if we replaced calls `Static.String()` with `Static.EncodeToString(true)` in all places where we actually need the quoted version and change `Static.String()` such that it returns the original / unquoted string. And finally replace `Static.EncodeToString(false)` with `Static.String()`.
However, this PR is already large enough and it's probably better to do this in a separate PR.

Reply:
Static.String() returns quoted to support traceql query roundtrip/normalization.  This is done through Stringer interface (recursively through the AST)